js/EmbedContentRequest.js

```diff
@@ -15,6 +15,65 @@ export const SCHEMA = {
       "description": "Language code from https://github.com/learningequality/le-utils/blob/main/le_utils/resources/languagelookup.json",
       "pattern": "^[a-z]{2,3}(?:-[a-zA-Z]+)?$"
     },
+    "url": {
+      "type": "string",
+      "pattern": "^(https?:\\/\\/(?:storage\\.cloud\\.google\\.com|localhost(?::[0-9]+)?)\\/[a-z0-9-._~!$&'()*+,;=:@%\\/\\?]+)$"
+    },
+    "preset": {
+      "type": "string",
+      "description": "Presets from https://github.com/learningequality/le-utils/blob/main/le_utils/constants/format_presets.py",
+      "enum": [
```

Comment:
Didnt know if there is a better way of generating this enum values

Reply:
Maybe categorize them?  @AlexVelezLl

Reply:
Not really sure @bjester?

Reply:
So we have done something like this before. Here's the example:
- definitions: https://github.com/learningequality/le-utils/blob/main/spec/schema-mastery_criteria.json
- usage: https://github.com/learningequality/le-utils/blob/main/spec/schema-completion_criteria.json#L20

In the [`schema-mastery_criteria.json`](https://github.com/learningequality/le-utils/blob/main/spec/schema-mastery_criteria.json), you'll notice the [`$exportConstants` key](https://github.com/learningequality/le-utils/blob/main/spec/schema-mastery_criteria.json#L11) which causes the generated files to export those. This approach also requires a specific implementation for the validators, like [this](https://github.com/learningequality/studio/blob/unstable/contentcuration/contentcuration/frontend/shared/leUtils/CompletionCriteria.js#L10).

That said, it would require some refactoring that I think is out of scope for our changes here. Optionally, we could also limit the presets provided here. For instance, the `*_thumbnail` presets we can omit, along with `slideshow_manifest`, `exercise_graphie`, and `exercise_image`. But that isn't a big deal. Studio should definitely not send those as they're not useful. So having the API schema restrict that isn't terrible either.

Reply:
Thanks @bjester. Heads up @AlexVelezLl  that the change is made in `spec/schema-embed_content_request.json` not this file